pkg/openshiftService/openshiftService.go

```diff
@@ -0,0 +1,57 @@
+package openshiftService
+
+import (
+	userClientV1Api "github.com/openshift/api/user/v1"
+	"github.com/patrickmn/go-cache"
+	"time"
+)
+
+type OpenshiftService struct {
+	openshiftClient openshiftClientInterface
+	cache           *cache.Cache
+}
+
+const groupsCacheKey = "groups"
+
+func NewOpenshiftService() (*OpenshiftService, error) {
+	openshiftClient, err := newOpenshiftClient()
+	if err != nil {
+		return nil, err
+	}
+	return &OpenshiftService{
+		openshiftClient: openshiftClient,
+		cache:           cache.New(1*time.Minute, 10*time.Minute),
+	}, nil
+}
+
+func (oc *OpenshiftService) GetGroupsUserBelongsTo(username string) ([]string, error) {
+	groups, err := oc.getGroupsWithCache()
+	if err != nil {
+		return nil, err
+	}
+
+	groupsUserBelongsTo := make([]string, 0)
+	for _, group := range groups.Items {
+		for _, user := range group.Users {
+			if user == username {
+				groupsUserBelongsTo = append(groupsUserBelongsTo, group.Name)
+				break
+			}
+		}
+	}
+
+	return groupsUserBelongsTo, nil
+}
+
+func (oc *OpenshiftService) getGroupsWithCache() (*userClientV1Api.GroupList, error) {
+	valueFromCache, found := oc.cache.Get(groupsCacheKey)
+	if found {
+		return valueFromCache.(*userClientV1Api.GroupList), nil
```

Comment:
syntax question, `valueFromCache.(*userClientV1Api.GroupList)` . the dot in the middle meaning that `valueFromCache` access the type of `(*userClientV1Api.GroupList)` ?

Reply:
type convertion